blocks.js

```diff
@@ -468,12 +468,9 @@ function attachTypeSwitcherActions() {
 	} );
 
 	Object.keys( typeToTag ).forEach( function( type ) {
-		var selector = '.switch-block__block .type-icon-' + type;
-		var button = queryFirst( selector );
-		var label = queryFirst( selector + ' + label' );
-
+		var iconSelector = '.switch-block__block .type-icon-' + type;
+		var button = queryFirst( iconSelector ).parentNode;
```

Comment:
Ah, at first I was going to suggest changing `iconSelector` to remove ` .type-icon-' + icon` child selector, to avoid accessing parent node, but I see why it's needed to ensure that each icon has its own event bound. The alternative might be to bind a single event handler to the `.switch-block__block` button and determine at the time of click which icon type it contains. But this seems to work well and is a much more direct change 👍